operator/pkg/helm/helm.go

```diff
@@ -113,13 +113,26 @@ func renderChart(namespace, values string, chrt *chart.Chart) (string, error) {
 	if err != nil {
 		return "", err
 	}
-
 	files, err := engine.Render(chrt, vals)
-	crdFiles := chrt.CRDObjects()
 	if err != nil {
 		return "", err
 	}
-
+	crdFiles := chrt.CRDObjects()
+	if chrt.Metadata.Name == "base" {
+		base, _ := valuesMap["base"].(map[string]interface{})
+		enableCRDs, _ := base["enableCRDs"].(bool)
```

Comment:
This is suspicious to me to take such a strong dependency on values here, as well as the `crd-all.gen.yaml` name..

@ostromart @richardwxn ?

Reply:
the reason is we need to install the crd-operator.yaml in all cases..
and current operator will install all crds under `crd` folder for all charts.

Reply:
@ostromart @richardwxn any better solutions than this? or can we go ahead with current way?  Basically what we need here is using `enableCRDs` to control whether we will install `crd-all.gen.yam` but keep `crd-operator` always being insalled. Thanks!

Reply:
in general we should avoid adding things to values.yaml as this is an unreviewed API - API changes should go through istio/api. 
you could consider moving the crds out of base and into some other component if they are essential to that component's operation.

Reply:
Thanks @ostromart  but copy to a new component is really heavy..   based on that,  can we go ahead with current solution?

Reply:
why is it heavy? and why would it be less preferred than adding something to a backdoor api we are trying to eliminate?

Reply:
copy to existing component can not solve the problem as we need a way to disable part of the crds. eg: for pilot, I need a way to control whether I will install crd-all.gen.yaml  + crd-operator.yaml   or only crd-operator.yaml.

Reply:
removed the change in helm.go here instead I am make the crd file under templates which we can have flag set